django/test/utils.py

```diff
@@ -113,21 +113,25 @@ def setup_test_environment(debug=None):
     if debug is None:
         debug = settings.DEBUG
 
-    _SavedSettings.debug = settings.DEBUG
-    settings.DEBUG = debug
+    # Storing previous values in the settings module itself is problematic.
+    # See #20636.  We store them in a private class instead.  Also, we use
+    # a Namespace object for convenience because it creates new attributes
+    # on assignment.
+    saved_data = Namespace()
```

Comment:
Is there some advantage to using `Namespace` compared to `saved_data = _TestState()`?